### PR TITLE
GEOMESA-2291 CQEngine - handle filter functions, etc

### PR DESCRIPTION
**Pull request by jnh5y**
* Handles the cases around (in)equalities.

Signed-off-by: Jim Hughes <jnh5y@ccri.com>

### CODE REVIEW

File: geomesa-memory/geomesa-cqengine/src/test/scala/org/locationtech/geomesa/memory/cqengine/utils/SampleFeatures.scala [hunk 1]
```diff
@@ -259,4 +259,13 @@ object SampleFilters {
     "Who ILIKE 'ADD%'",
     "Who ILIKE '%DA%'"
   )
+
+  val functionPredicates: Seq[Filter] = Seq(
+    "strConcat(Who, What) = 'Addams1'",
```
Comment (elahrvivaz) [inline geomesa-memory/geomesa-cqengine/src/test/scala/org/locationtech/geomesa/memory/cqengine/utils/SampleFeatures.scala]: will this work with multiple AND'd predicates?

Comment (elahrvivaz) [reply geomesa-memory/geomesa-cqengine/src/test/scala/org/locationtech/geomesa/memory/cqengine/utils/SampleFeatures.scala]: can you add some tests for AND'd predicates of various types?

**Comment by jnh5y** [comment]
I didn't check it in here... but if you want more tests (and some non-determinism to boot!), you can check out this commit: https://github.com/ccri/geomesa/commit/039f486526538f40c78a1203316499e77020bf02.

**Comment by elahrvivaz** [comment]
@jnh5y do those extended tests pass? is the non-determinism an issue, or think it's ok to merge?

**Comment by jnh5y** [comment]
@elahrvivaz the extended tests pass.  The non-determinism comes from the fact that I was generating a random subset of 3-combinations of all the existing filters....;)

This should be good to go.

**Pull request merged**